.eslintrc

```diff
@@ -9,7 +15,26 @@
     "no-prototype-builtins": "off",
 
     // Replaced by TypeScript's static checking.
-    "react/prop-types": "off"
+    "react/prop-types": "off",
+
+    // Upgrade TS rules from warning to error.
+    "@typescript-eslint/no-unused-vars": "error",
+
+    // Disable TS rules that we dislike.
```

Comment:
We have code violating each of the rules below. "We dislike" means it is my personal view that we should turn off the rule rather than trying to change our code, as I think we have legitimate uses for these patterns.